core/src/main/java/org/hibernate/ogm/massindex/impl/OptionallyWrapInJTATransaction.java

```diff
@@ -130,21 +114,6 @@ public void consume(TupleSupplier supplier) {
 		}
 	}
 
-	private void consumeInTransaction(Tuple tuple) {
-		TransactionManager transactionManager = getTransactionManager();
-		try {
-			final Session session = factory.openSession();
-			transactionManager.begin();
-			delegate.run( session, tuple );
-			transactionManager.commit();
-			session.close();
-		}
-		catch ( Throwable e ) {
-			errorHandler.handleException( log.massIndexerUnexpectedErrorMessage(), e );
-			rollback( transactionManager, e );
-		}
-	}
-
 	private void consumeInTransaction(TupleSupplier supplier) {
```

Comment:
I think I have mentioned it before but `TuplesSupplier` would be better as it supplies an iterator with `Tuples`.